### ISSUE TITLE
use mongodb with swoole together will occurs error

### ISSUE THREAD

**Issue by jonnywang** [mongodb/mongo-php-driver#213]
i use swoole (https://github.com/swoole/swoole-src) , mongodb together, test code like 
```php
<?php

$server = new swoole_server('127.0.0.1', 9501);
$server->set(array(
    'worker_num' => 1,
    'task_worker_num' => 2,
    'log_file'   => './swoole.log'
));

$server->on('workerStart', function($serv, $worker_id){
    echo 'worker start :' . $worker_id . ',pid :' . getmypid() . PHP_EOL;
});

$server->on('task', function($serv, $task_id, int $from_id, string $data){
    echo 'on task  :' . $task_id . ',pid :' . getmypid() . PHP_EOL;
});

$server->on('finish', function($serv, $task_id, string $data){
    echo 'on finish  :' . $task_id . ',pid :' . getmypid() . PHP_EOL;
});

$server->on('receive', function ($serv, $fd, $from_id, $data) {
    $serv->reload(true);
});

$server->start();
```

where server receive any data, i will set server to reload, but occurs

```
[Mon Feb  1 17:12:39 2016]  Script:  '/Users/xingqiba/Desktop/haha.php'
---------------------------------------
/Users/xingqiba/data/softs/mongodb-1.1.2/php_phongo.c(2248) : Block 0x7fe95255a040 status:
Beginning:  	Freed (magic=0x00000000, expected=0x99954317)
    Start:	Overflown (magic=0x00000035 instead of 0x4060D864)
          	At least 4 bytes overflown
worker start :1,pid :71250
[Mon Feb  1 17:12:39 2016]  Script:  '/Users/xingqiba/Desktop/haha.php'
---------------------------------------
/Users/xingqiba/data/softs/mongodb-1.1.2/php_phongo.c(2248) : Block 0x7fe95255a040 status:
Beginning:  	Freed (magic=0x00000000, expected=0x99954317)
    Start:	Overflown (magic=0x00000035 instead of 0x4060D864)
          	At least 4 bytes overflown
```

if i don't use mongodb in php.ini, it's ok!

help

**Comment by bjori** [comment]
Thank you for the report!
I relayed this ticket to our JIRA: https://jira.mongodb.org/browse/PHPC-555 for better tracking.

I haven't tried your reproduce case yet, but it doesn't appear to be using the MongoDB extension so I'm curious if you can share you configuration? Or does this always happen using the example above as long as the MongoDB extension is loaded? That seems a bit strange :)

The error log you attached definitely indicates a problem in the MongoDB extension

**Comment by jonnywang** [comment]
environment:
mac 10.11.2
php 7.0.2 7.0.0 5.5.28
swoole  git master

swoole configure
```
phpize
./configure --with-php-config=x   --enable-    sockets --enable-openssl --enable-swoole --enable-async-redis --enable-async-mysql --enable-async-httpclient --enable-jemalloc
make 
make install
```

```php
php7.0.2  haha.php  //or php5.5.28 haha.php
```

```sh
telnet 127.0.0.1 9501
type any words
```
php7.0.2 
```
worker start :0,pid :74618
worker start :1,pid :74619
worker start :2,pid :74620
zend_mm_heap corrupted
zend_mm_heap corrupted
worker start :1,pid :74629
worker start :2,pid :74630
```

php5.5.28
```
worker start :1,pid :74693
worker start :0,pid :74692
worker start :2,pid :74694
[Tue Feb  2 09:35:55 2016]  Script:  '/Users/xingqiba/Desktop/haha.php'
---------------------------------------
/Users/xingqiba/data/softs/mongodb-1.1.2/php_phongo.c(2248) : Block 0x7fa50ac02d20 status:
Invalid pointer: ((size=0x0000716e) != (next.prev=0x00007fa5))
worker start :1,pid :74698
[Tue Feb  2 09:35:55 2016]  Script:  '/Users/xingqiba/Desktop/haha.php'
---------------------------------------
/Users/xingqiba/data/softs/mongodb-1.1.2/php_phongo.c(2248) : Block 0x7fa50ac02d20 status:
Invalid pointer: ((size=0x0000716e) != (next.prev=0x00007fa5))
worker start :2,pid :74699
```
it always happen using the example above as long as the MongoDB extension is loaded.

php.ini
```
extension=mongodb.so
extension=swoole.so
```
php7.0.2 configure
```
./configure  --prefix=/Users/xingqiba/data/server/php7.0.2 --with-config-file-path=/Users/xingqiba/data/server/php7.0.2/etc --enable-mysqlnd --with-mysqli --with-iconv=/usr/local/Cellar/libiconv/1.14 --with-freetype-dir --with-jpeg-dir=/usr --with-png-dir --with-zlib --with-libxml-dir=/usr --enable-xml --disable-rpath --enable-bcmath --enable-shmop --enable-inline-optimization --with-curl=/usr/local/opt/curl --enable-mbregex --enable-fpm --enable-mbstring --with-mcrypt --with-gd --enable-gd-native-ttf --with-openssl=/usr/local/Cellar/openssl/1.0.2d_1 --with-mhash --enable-pcntl --enable-sockets --with-xmlrpc --enable-zip --enable-soap --enable-sysvshm --enable-sysvsem --enable-sysvmsg --without-bz2 --with-readline=/usr/local/opt/readline --with-libedit --with-pdo-mysql --with-config-file-scan-dir=/Users/xingqiba/data/server/php7.0.2/etc/ext.d/ --enable-debug LDFLAGS=-L/usr/local/opt/bison/lib YACC=/usr/local/Cellar/bison/3.0.4/bin/bison
```

php5.5.28 configure
```
./configure  --prefix=/Users/xingqiba/data/server/php5.5.28 --with-config-file-path=/Users/xingqiba/data/server/php5.5.28/etc --enable-mysqlnd --with-mysqli --with-iconv=/usr --with-freetype-dir --with-jpeg-dir=/usr --with-png-dir --with-zlib --with-libxml-dir=/usr --enable-xml --disable-rpath --enable-bcmath --enable-shmop --enable-inline-optimization --with-curl=/usr/local/opt/curl --enable-mbregex --enable-fpm --enable-mbstring --with-mcrypt --with-gd --enable-gd-native-ttf --with-openssl --with-mhash --enable-pcntl --enable-sockets --with-xmlrpc --enable-zip --enable-soap --enable-sysvshm --enable-sysvsem --enable-sysvmsg --without-bz2 --with-readline=/usr/local/opt/readline --with-libedit --with-pdo-mysql --with-config-file-scan-dir=/Users/xingqiba/data/server/php5.5.28/etc/ext.d/ --enable-debug
```

**Comment by jmikola** [comment]
I've since released [1.1.5](https://github.com/mongodb/mongo-php-driver/releases/tag/1.1.5), which I believe may fix this issue if it is a duplicate of #209. It's possible that the backtraces differ as a result of using jemalloc instead of Zend's memory manager; however, I expect the root cause may be the same. I will leave it open for you to confirm.

Note that the patch is only able to resolve the problem when libmongoc is bundled (i.e. statically compiled) with the driver. If libmongoc is loaded as a system library, that will rely on an upstream fix (see: [CDRIVER-1165](https://jira.mongodb.org/browse/CDRIVER-1165)).

**Comment by jonnywang** [comment]
thanks, fixed with 1.1.5
